app/helpers/protocols_io_helper.rb

```diff
@@ -285,6 +285,49 @@ def protocols_io_fill_desc(json_hash)
     description_string
   end
 
+  def protocols_io_return_first_step_guid(unordered_step_json)
+    return_step = 'null'
+    unordered_step_json['steps'].each do |step|
+      return_step = step unless step['previous_guid']
+    end
+    return_step
+  end
+
+  def protocols_io_count_steps(step_json)
+    count = 0
+    step_json['steps'].each do |step|
+      count += 1
+    end
+    count
+  end
+
+  def protocols_io_guid_reorder_step_json(unordered_step_json)
+    base_step = protocols_io_return_first_step_guid(unordered_step_json)
```

Comment:
You can find the base step like this:
`unordered_step_json['steps'].find{|step| step['previous_guid'].nil?}`